src/backend/distributed/executor/executor.c

```diff
@@ -388,8 +390,12 @@ CitusExecScan(CustomScanState *node)
 		bool interTransactions = false;
 		int targetPoolSize = DEFAULT_POOL_SIZE;
 
-		/* we are taking locks on partitions of partitioned tables */
-		LockPartitionsInRelationList(distributedPlan->relationIdList, AccessShareLock);
+		/*
+		 * PostgreSQL takes locks on all partitions in the executor. It's not entirely
+		 * clear why this is necessary (instead of locking the parent during DDL), but
+		 * We do the same for consistency.
+		 */
+		LockPartitionsForDistributedPlan(distributedPlan);
```

Comment:
Just to make sure: I think for DDLs we expect Postgres to acquire the locks on the utility hook so we don't care about them on master branch or here as well. Can you please confirm that?

Reply:
Correct.